vaadin-context-menu-flow-parent/vaadin-context-menu-flow/src/main/java/com/vaadin/flow/component/contextmenu/MenuItemBase.java

```diff
@@ -186,6 +192,37 @@ public boolean isChecked() {
         return getElement().getProperty("_checked", false);
     }
 
+    /**
+     * Sets the keep open state of this menu item. An item that marked as keep
+     * open prevents menu from closing when clicked.
+     *
+     * @param keepOpen
+     *            {@code true} to enable keeping the menu open when toggle this
+     *            item selection, {@code false} to disable it.
+     */
+    public void setKeepOpen(boolean keepOpen) {
+        if (keepOpen && isParentItem()) {
```

Comment:
Would it be possible to make a unit test for this check?

Reply:
Added a test.